### PR TITLE
WT-12402 Add stats to track when trees are skipped in eviction server walks

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: dist/stat_data.py [hunk 1]
```diff
@@ -326,10 +326,16 @@ def __init__(self, name, desc, flags=''):
     CacheStat('cache_eviction_queue_empty', 'eviction server candidate queue empty when topping up'),
     CacheStat('cache_eviction_queue_not_empty', 'eviction server candidate queue not empty when topping up'),
     CacheStat('cache_eviction_server_evicting', 'eviction server evicting pages'),
-    CacheStat('cache_eviction_server_slept', 'eviction server slept, because we did not make progress with eviction'),
+    CacheStat('cache_eviction_server_skip_checkpointing_trees', 'skip trees that are being checkpointed'),
```
Comment (Siddhartha8899) [inline dist/stat_data.py:329]: could we begin the description of these stats with `eviction server`?

File: dist/stat_data.py [hunk 2]
```diff
@@ -326,10 +326,16 @@ def __init__(self, name, desc, flags=''):
...
+    CacheStat('cache_eviction_server_skip_dirty_pages_during_checkpoint', 'skip dirty pages during a running checkpoint'),
     CacheStat('cache_eviction_server_skip_pages_retry', 'skip pages that previously failed eviction and likely will again'),
     CacheStat('cache_eviction_server_skip_pages_last_running', 'skip pages that are written with transactions greater than the last running'),
-    CacheStat('cache_eviction_server_skip_dirty_pages_during_checkpoint', 'skip dirty pages during a running checkpoint'),
+    CacheStat('cache_eviction_server_skip_metatdata_with_history', 'skip metadata pages with history'),
+    CacheStat('cache_eviction_server_skip_trees_not_useful_before', 'skip trees that are not useful before'),
```
Comment (etienneptl) [inline dist/stat_data.py:334]:
```suggestion
    CacheStat('cache_eviction_server_skip_trees_not_useful_before', 'skip trees that were not useful before'),
```

**Review by etienneptl — APPROVED** [review]
One comment that does not need another review, will love those stats, thank you!

File: dist/stat_data.py (hunk 1 above)
Comment (quchenhao) [reply dist/stat_data.py]: done

File: dist/stat_data.py (hunk 2 above)
Comment (quchenhao) [reply dist/stat_data.py]: fixed

**Review by Siddhartha8899 — APPROVED** [review]

**Pull request merged**